registrations/metrics.py

```diff
@@ -233,6 +271,90 @@ def registrations_change_messaging_total_last(self, start, end):
             .filter(action='change_messaging')\
             .count()
 
+    def optout_msg_type_sum(self, msg_type, start, end):
+        result = utils.search_optouts({
+            "created_at__gt": start,
+            "created_at__lte": end,
+        })
+
+        identities = []
+        for data in result:
+            identities.append(data['identity'])
+
+        return Registration.objects.filter(
+                    mother_id__in=identities,
+                    data__msg_type=msg_type).count()
+
+    def optout_msg_type_total_last(self, msg_type, start, end):
+        result = utils.search_optouts({
+            "created_at__lte": end,
+        })
+
+        identities = []
+        for data in result:
+            identities.append(data['identity'])
+
+        return Registration.objects.filter(
+                    mother_id__in=identities,
+                    data__msg_type=msg_type).count()
+
+    def optout_receiver_type_sum(self, receiver_type, start, end):
+        result = utils.search_optouts({
+            "created_at__gt": start,
+            "created_at__lte": end,
+        })
+
+        identities = []
+        for data in result:
+            identities.append(data['identity'])
+
+        return Registration.objects.filter(
+                    mother_id__in=identities,
+                    data__msg_receiver=receiver_type).count()
+
+    def optout_receiver_type_total_last(self, receiver_type, start, end):
+        result = utils.search_optouts({
+            "created_at__lte": end,
+        })
+
+        identities = []
+        for data in result:
+            identities.append(data['identity'])
+
+        return Registration.objects.filter(
+                    mother_id__in=identities,
+                    data__msg_receiver=receiver_type).count()
+
+    def optout_reason_sum(self, reason, start, end):
+        result = utils.search_optouts({
+            "reason": reason,
+            "created_at__gt": start,
+            "created_at__lte": end,
+        })
+        return len(list(result))
```

Comment:
Same here for using `sum(1 for r in result)` instead of `len(list(result))`, and the ones below this